mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster.go

```diff
@@ -1243,7 +1243,7 @@ func ResourceDataprocCluster() *schema.Resource {
 								Schema: map[string]*schema.Schema{
 									"kerberos_config": {
 										Type:        schema.TypeList,
-										Required:    true,
+										Optional:    true,
```

Comment:
I believe `kerberos_config` was marked as required initially since this is the only subfield under `security_config`. We'd generally want to prevent sending empty blocks, like `security_config{}`, unless explicitly sending `security_config{}` to the API is different from not sending the field at all. 

Do we want to allow users to explicitly specify `security_config{}` in this case?

Reply:
> I believe kerberos_config was marked as required initially since this is the only subfield under security_config

That is true historically. But other fields have been added since then. 

> We'd generally want to prevent sending empty blocks, like security_config{}, unless explicitly sending security_config{} to the API is different from not sending the field at all.
>
> Do we want to allow users to explicitly specify security_config{} in this case?

I don't think Dataproc's service will differentiate between an empty and an absent `security_config`. 

The flatten side is what prompted this PR, as there are in-development hidden fields being added to `security_config`. Dataproc would return `security_config{}`, which terraform would try to flatten, resulting in errors. I made the change to the schema mostly for consistency.